src/Main.hs

```diff
@@ -10,23 +10,43 @@ import           Control.Monad.State            ( StateT(runStateT) )
 import           System.FilePath                ( takeDirectory
                                                 , replaceExtension
                                                 )
+import           System.FilePath.Posix          ( splitFileName )
 import           System.Directory               ( createDirectoryIfMissing )
 import           Path
 import           AST
 
 import           Infer.Solve
 import           Infer.Infer
+import           Options.Applicative
+import           Constants.Flags
+import           Tools.CommandLine
 import           Compile
 import qualified AST.Solved                    as Slv
 import qualified Explain.Format                as Explain
 
+
 main :: IO ()
-main = do
-  entrypoint <- head <$> getArgs
-  astTable   <- buildASTTable entrypoint
-  putStrLn $ ppShow astTable
 
+-- >>>>>>>>>>>>>>>>>>>>
+-- main = do
+--   entrypoint <- head <$> getArgs
+-- ====================
+main = run =<< execParser opts
+ where
+  opts = info
+    (parseTransform <**> helper)
+    (fullDesc <> progDesc "swap insanity with _______" <> header hashBar)
+
+run :: TransformFlags -> IO ()
+run (TransformFlags i o c) = do
+  let entrypoint = show i
```

Comment:
That will allow you to make it run:
```suggestion
  let (FileInput entrypoint) = i
```

Although I believe that input could also be a `StdInput`. In that case we would need to pattern match against it, and retrieve the source one way or another.

Reply:
Ahhhhh! That is much more straightforward than what I thought the solution was. Is this an example of pattern-matching to pull stuff out of `i`? (oder ?)

Reply:
Yes precisely. That’s mostly how you access values inside adts. Like you would with a maybe or either.